airflow/www/security.py

```diff
@@ -520,6 +520,17 @@ def _has_perm(self, action_name, resource_name):
         self._get_and_cache_perms()
         return (action_name, resource_name) in self.perms
 
+    def has_all_dags_edit_access(self):
+        """
+        Has all the dag access in any of the 3 cases:
+        1. Role needs to be in (Admin, Viewer, User, Op).
+        2. Has can_read action on dags resource.
+        3. Has can_edit action on dags resource.
+        """
+        return self._has_role(['Admin', 'Op', 'User']) or self._has_perm(
```

Comment:
Why do we need the rule check? Don't the roles already have this permission? (And by adding the Role is "x" it means the permissions on the build in roles don't have any effect)

Reply:
I agree. I didn't like the `_has_role` item in there either.
I think it's done for performance reasons.
Do you want me to remove it?

Reply:
@Jorricks, I also think we should remove the role check, and as @jhtimmins mentioned probably this helper too?

Reply:
Alright. Pushing an update that removes this unnecessary helper function.
I originally implemented it for performance reasons, as many of these helpers seem to be, but I agree that the performance improvement is negligible and not worth the extra maintenance.